service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureStorageAccessServiceUnitTest.java

```diff
@@ -0,0 +1,278 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class AzureStorageAccessServiceUnitTest extends BaseUnitTest {
+
+  private final OffsetDateTime startTime = OffsetDateTime.now().minusMinutes(15);
+  private final OffsetDateTime expiryTime = OffsetDateTime.now().plusMinutes(60);
+
+  private ControlledAzureStorageResource buildStorageAccount() {
+    return new ControlledAzureStorageResource(
+        UUID.randomUUID(),
+        UUID.randomUUID(),
+        "fake",
+        "description",
+        CloningInstructions.COPY_NOTHING,
+        null,
+        PrivateResourceState.NOT_APPLICABLE,
+        AccessScopeType.ACCESS_SCOPE_SHARED,
+        ManagedByType.MANAGED_BY_USER,
+        null,
+        "fake",
+        "us-east1");
+  }
+
+  private ControlledAzureStorageContainerResource buildStorageContainerResource(
+      PrivateResourceState privateResourceState,
+      AccessScopeType accessScopeType,
+      ManagedByType managedByType) {
+    return new ControlledAzureStorageContainerResource(
+        UUID.randomUUID(),
+        UUID.randomUUID(),
+        "fake",
+        "description",
+        CloningInstructions.COPY_NOTHING,
+        null,
+        privateResourceState,
+        accessScopeType,
+        managedByType,
+        null,
+        UUID.randomUUID(),
+        "fake");
+  }
+
+  private void assertValidToken(String sas, BlobContainerSasPermission expectedPermissions) {
+    Pattern protocolRegex = Pattern.compile("spr=https&");
+    // SAS tokens start and expiry times are UTC
+    Pattern startTimeRegex =
+        Pattern.compile(
+            "st="
+                + startTime
+                    .atZoneSameInstant(ZoneOffset.UTC)
+                    .format(DateTimeFormatter.ISO_LOCAL_DATE));
+    Pattern expiryTimeRegex =
+        Pattern.compile(
+            "se="
+                + expiryTime
+                    .atZoneSameInstant(ZoneOffset.UTC)
+                    .format(DateTimeFormatter.ISO_LOCAL_DATE));
+    Pattern signedResourceRegex = Pattern.compile("sr=c&");
+    Pattern permissionsRegex = Pattern.compile("sp=" + expectedPermissions.toString() + "&");
+
+    assertThat("SAS is https", protocolRegex.matcher(sas).find());
+    assertThat("SAS validity starts today", startTimeRegex.matcher(sas).find());
+    assertThat("SAS validity ends today", expiryTimeRegex.matcher(sas).find());
+    assertThat("SAS is for a container resource", signedResourceRegex.matcher(sas).find());
+    assertThat("SAS grants correct permissions", permissionsRegex.matcher(sas).find());
+  }
+
+  @Test
+  public void createAzureStorageContainerSasToken_readonly() throws InterruptedException {
+    var mockSam = mock(SamService.class);
```

Comment:
For mocking out Spring beans, you can use the `@MockBean` annotation to avoid repeating this setup at the start of every test (e.g. https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/test/java/bio/terra/workspace/service/workspace/GcpCloudContextUnitTest.java#L42)
It also helps by having the services your tests depends on listed at the top of the suite as either mocked (`@MockBean`) or not (`@Autowired`)

Reply:
👍 will do